### DISCUSSION TITLE
Does GitHub deliberately block indexing of code in robots.txt?

### DISCUSSION THREAD

**Discussion by simonw** [community/community#20958] [Other Feature Feedback, Questions, & Ideas]
https://github.com/robots.txt includes this:

```
User-agent: *

Disallow: /*/pulse
Disallow: /*/tree/
```
I host documentation on GitHub for some of my projects, for example this page: https://github.com/simonw/datasette-graphql/blob/main/examples/variables.md

But... that page can only be navigated to using the following path:

- https://github.com/simonw/datasette-graphql - allowed by `robots.txt`
- https://github.com/simonw/datasette-graphql/tree/main/examples - DISALLOWED by `robots.txt`
- https://github.com/simonw/datasette-graphql/blob/main/examples/variables.md - allowed by `robots.txt` - but the crawler won't get there because the `/tree/main/examples` page was disallowed

Is this a deliberate GitHub policy, or is it a bug in the `robots.txt` file?

**Comment by tobiasvl** [comment]
It's clearly deliberate, as you can see in `robots.txt`:

> \# If you would like to crawl GitHub contact us via https://support.github.com?tags=dotcom-robots
> \# We also provide an extensive API: https://docs.github.com

In other words, GitHub doesn't want the code to be crawled. If someone wants to index GitHub code, they should contact GitHub Support, or use the API.

Could GitHub keep disallowing crawling of the code, but specifically allow `.md` files in the repository to be crawled? I guess so. That would fix your issue, and you could suggest that.

But the main issue here is that you're "misusing" GitHub: You're putting documentation in the code repository, where it doesn't really belong. It would be more appropriate to:

* Put the documentation at https://datasette-graphql-demo.datasette.io – perhaps autogenerated by [sphinx](https://www.sphinx-doc.org/en/master/)
* Link to the documentation in a [SUPPORT file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-support-resources-to-your-project)
* Document your project in the [wiki](https://docs.github.com/en/communities/documenting-your-project-with-wikis), which _is_ crawlable

**Reply by simonw** [reply]
Ideally I'd like to see GitHub document this somewhere. I'm sure I'm not the only person who has assumed that markdown files hosted on GitHub will be indexed by Google - especially since the `README.md` for a project IS indexed.

**Reply by iiasceri** [reply]
Saw this on gists when I prompted OpenAI's (closedAI) GPT and refused saying robots.txt does not allow it.
